presto-main/src/main/java/com/facebook/presto/operator/scalar/RowIndeterminateOperator.java

```diff
@@ -0,0 +1,174 @@
+import static com.facebook.presto.sql.gen.InvokeFunctionBytecodeExpression.invokeFunction;
+import static com.facebook.presto.sql.gen.SqlTypeBytecodeExpression.constantType;
+import static com.facebook.presto.type.UnknownType.UNKNOWN;
+import static com.facebook.presto.util.CompilerUtils.defineClass;
+import static com.facebook.presto.util.CompilerUtils.makeClassName;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.bytecode.Access.FINAL;
+import static io.airlift.bytecode.Access.PUBLIC;
+import static io.airlift.bytecode.Access.STATIC;
+import static io.airlift.bytecode.Access.a;
+import static io.airlift.bytecode.Parameter.arg;
+import static io.airlift.bytecode.ParameterizedType.type;
+import static io.airlift.bytecode.expression.BytecodeExpressions.constantFalse;
+import static io.airlift.bytecode.expression.BytecodeExpressions.constantInt;
+
+public class RowIndeterminateOperator
+        extends SqlOperator
+{
+    public static final RowIndeterminateOperator ROW_INDETERMINATE = new RowIndeterminateOperator();
+
+    private RowIndeterminateOperator()
+    {
+        super(INDETERMINATE, ImmutableList.of(withVariadicBound("T", "row")), ImmutableList.of(), BOOLEAN.getTypeSignature(), ImmutableList.of(parseTypeSignature("T")));
+    }
+
+    @Override
+    public ScalarFunctionImplementation specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        checkArgument(arity == 1, "Expected arity to be 1");
+        Type type = boundVariables.getTypeVariable("T");
+        Class<?> indeterminateOperatorClass = generateIndeterminate(type, functionRegistry);
+        MethodHandle methodHandle = methodHandle(indeterminateOperatorClass, "indeterminate", type.getJavaType(), boolean.class);
+        return new ScalarFunctionImplementation(
+                false,
+                ImmutableList.of(valueTypeArgumentProperty(USE_NULL_FLAG)),
+                methodHandle,
+                isDeterministic());
+    }
+
+    private static Class<?> generateIndeterminate(Type type, FunctionRegistry functionRegistry)
+    {
+        CallSiteBinder binder = new CallSiteBinder();
+
+        ClassDefinition definition = new ClassDefinition(
+                a(PUBLIC, FINAL),
+                makeClassName("RowIndeterminateOperator"),
+                type(Object.class));
+
+        Parameter value = arg("value", type.getJavaType());
+        Parameter isNull = arg("isNull", boolean.class);
+
+        MethodDefinition method = definition.declareMethod(
+                a(PUBLIC, STATIC),
+                "indeterminate",
+                type(boolean.class),
+                value,
+                isNull);
+
+        Scope scope = method.getScope();
+        BytecodeBlock body = method.getBody();
+
+        Variable wasNull = scope.declareVariable(boolean.class, "wasNull");
```

Comment:
What's the use of `wasNull` variable?

Reply:
It is needed for the `invokeFunction` generator. Although that variable is always null because of the `isNull` parameter convention, the variable itself is needed. See https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeUtils.java#L238.

Maybe we should skip that conditionally, but than we need conditionally reset that variable to `false` only if the one exist. And currently scope does not allow you to check if the variable is declared.

Nevertheless i removed all the places where the `wasNull` variable was reset to static false.